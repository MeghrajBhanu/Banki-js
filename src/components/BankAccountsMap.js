import {useState} from 'react'
import Modal from 'react-modal';
import { useNavigate } from 'react-router';
import "../pages/Landing.css"
const customStyles = {
    content: {
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      marginRight: '-50%',
      transform: 'translate(-50%, -50%)',
    },
  };
  

const BankAccountsMap = ({
  id,
  name,
  email,
  pancard,
  account_type,
  fd,
  bankName,
  balance,
  
}) => {
    const Navigate=useNavigate();
    
    const handleId=(e)=>{
        
       
        Navigate("/login/landing/"+id)
    }
    
    
  return (
    <div className="bank-card" >
    {/*<div class="col-sm-3 mt-2 mb-2 mx-5 p-1" ></div>*/}
      
      {/* <Modal isOpen={isOpen} onRequestClose={closeModal} style={customStyles} ariaHideApp={false}>
        <button onClick={closeModal}>X</button>
        <h4>Your details associated with {bankName}</h4>
        <h6>Registerd Name:{name}</h6>

        <p>
            Email: {email}
            <br />
            FixedDeposits: {fd}
            <br />
            Total balance: {balance}
            <br />
            AccountType: {account_type}
            <br />
            AccountNumber: {id}
        </p>
        <h6>Pancard: {pancard}</h6>

        
      </Modal> */}
      
      <div className="card" style={{ width: "18rem" }}>
        <div className="card-body">
          <h5 className="card-title">{bankName}</h5>
          <h6 className="card-subtitle mb-2 text-muted">
            {account_type + " Account"}
          </h6>
          <p className="card-text ">
            FixedDeposits: {fd}
            <br />
            Total balance: {balance}
          </p>
          <button className="btn btn-secondary" onClick={handleId}>
                View More
          </button>
          
        </div>
      </div>
    </div>
  );
};
export default BankAccountsMap;
