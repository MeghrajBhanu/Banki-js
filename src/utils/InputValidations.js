export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}