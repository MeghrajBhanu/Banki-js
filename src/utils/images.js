import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;