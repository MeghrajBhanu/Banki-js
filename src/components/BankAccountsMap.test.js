import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../utils/test_helper";
import renderer from "react-test-renderer";
import BankAccountsMap from "./BankAccountsMap";

describe("BankAccountsMap", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<BankAccountsMap />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});