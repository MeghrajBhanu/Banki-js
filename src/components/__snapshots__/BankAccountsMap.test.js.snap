// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankAccountsMap snapshot test 1`] = `
<div
  className="card"
>
  <img
    alt="bankLogo"
  />
  <div
    className="card-body"
  >
    <div
      className="text-center"
    >
      <span
        className="h6 "
      />
      <span>
         --
      </span>
      <span
        className="h3-h"
      >
        undefined Account
      </span>
    </div>
    <p
      className="text-center"
    >
      FixedDeposits: 
      <br />
      Total balance: 
    </p>
    <button
      className="btn btn-secondary"
      name="view more"
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "95px",
        }
      }
    >
      View More
    </button>
  </div>
</div>
`;
